### PR TITLE
Add ability to set the number of connector threads and the buffer siz…

### LINKED ISSUE #393: blaze ServerBuilder should support a configurable number of selectors
A configurable buffer size wouldn't hurt. At a minimum, it shouldn't be fixed to [an arbitrary value of 12](https://github.com/http4s/http4s/blob/master/blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala#L147).

### PR DESCRIPTION
**Pull request by bryce-anderson**
…e for the blaze server.

closes #393

### CODE REVIEW

File: blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala [hunk 1]
```diff
@@ -22,11 +23,18 @@ import org.log4s.getLogger
 import scala.concurrent.duration._
 import scalaz.concurrent.{Strategy, Task}
 
+case class ConnectorConfig(isNio2: Boolean, connectorThreads: Int, bufferSize: Int) {
```
Comment (rossabaker) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala]: Is `idleTimeout` part of a connector config?  Maybe also SSL?

Comment (bryce-anderson) [inline blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala]: I think you're right.

**Comment by rossabaker** [comment]
Looks good, unless we want to shift more into config.  (Alternatively, flatten config and not change existing code.  Grouping increasingly complex configuration is hard.  This is one reason I'm still not thrilled with our server builders.)

**Comment by bryce-anderson** [comment]
I think you're right about keeping it flat: some of the distinctions become somewhat arbitrary. I'll probably flatten it and see how it looks.

**Comment by rossabaker** [comment]
:+1: pending Travis.

**Pull request merged**